segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScaleProcessor.java

```diff
@@ -54,100 +54,127 @@
+        if (w != null) {
+            w.close();
+            this.writer.set(null);
+        }
 
-    @VisibleForTesting
-    AutoScaleProcessor(AutoScalerConfig configuration, EventStreamClientFactory cf,
-                       ScheduledExecutorService maintenanceExecutor) {
-        this(configuration, maintenanceExecutor);
-        clientFactory.set(cf);
+        this.clientFactory.close();
+        this.cacheCleanup.cancel(true);
     }
 
-    private void bootstrapRequestWriters() {
-
-        CompletableFuture<Void> createWriter = new CompletableFuture<>();
-
+    private void bootstrapRequestWriters(EventStreamClientFactory clientFactory, ScheduledExecutorService executor) {
         // Starting with initial delay, in case request stream has not been created, to give it time to start
         // However, we have this wrapped in consumeFailure which means the creation of writer will be retried.
         // We are introducing a delay to avoid exceptions in the log in case creation of writer is attempted before
         // creation of requeststream.
-        maintenanceExecutor.schedule(() -> Retry.indefinitelyWithExpBackoff(100, 10, 10000,
-                e -> {
-                    log.warn("error while creating writer for requeststream");
-                    log.debug("error while creating writer for requeststream {}", e);
-                })
-                .runAsync(() -> {
-                    if (clientFactory.get() == null) {
-                        EventStreamClientFactory factory = null;
-                        if (configuration.isAuthEnabled()) {
-                            factory = EventStreamClientFactory.withScope(NameUtils.INTERNAL_SCOPE_NAME,
-                                    ClientConfig.builder().controllerURI(configuration.getControllerUri())
-                                                .trustStore(configuration.getTlsCertFile())
-                                                .validateHostName(configuration.isValidateHostName())
-                                                .build());
-                        } else {
-                            factory = EventStreamClientFactory.withScope(NameUtils.INTERNAL_SCOPE_NAME,
-                                    ClientConfig.builder().controllerURI(configuration.getControllerUri()).build());
-                        }
-                        clientFactory.compareAndSet(null, factory);
-                    }
+        executor.schedule(
+                () -> Retry.indefinitelyWithExpBackoff(100, 10, 10000, this::handleBootstrapException)
+                        .runInExecutor(() -> bootstrapOnce(clientFactory), executor),
+                10, TimeUnit.SECONDS);
+    }
+
+    private void handleBootstrapException(Throwable e) {
+        log.warn("error while creating writer for requeststream"); // Don't output stack trace.
+        log.debug("error while creating writer for requeststream", e); // Only output stack trace in debug mode.
```

Comment:
Should we use `isDebugEnabled`, like in:

```
if (log.isDebugEnabled()) {
   log.debug("error while creating writer for requeststream", e);
} else {
   log.warn("error while creating writer for requeststream");
}
```

Reply:
This is how the code was there before... But I think this makes sense. I fixed it.